sycl/test-e2e/ESIMD/api/saturation_smoke.cpp

```diff
@@ -7,7 +7,10 @@
 //===----------------------------------------------------------------------===//
 // TODO: esimd_emulator fails due to unimplemented 'half' type
 // XFAIL: esimd_emulator
-// RUN: %{build} -fsycl-device-code-split=per_kernel -o %t.out
+// TODO: remove fno-fast-math option once a compiler issue resulting
```

Comment:
For this test it is unclear what kind of compiler issue can cause fail on this test with -ffast-math.
It calls only 1 function 'saturate()', which is immediately lowered to __esimd_sat().
Isn't it more correct to say that the test itself has some issues, e.g. code verifying results on host needs some tuning?